src/core/Akka/Actor/Scheduler/HashedWheelTimerScheduler.cs

```diff
@@ -159,11 +158,15 @@ private void Run()
                 _startTime = 1;
             }
 
+            // we've already started - need to prevent duplicates.
+            if (_workerInitialized.IsSet)
+                return;
+            
             _workerInitialized.Signal();
 
             do
             {
-                var deadline = WaitForNextTick();
+                var deadline = await WaitForNextTick();
```

Comment:
Can probably use `IAsyncEnumerable` here.

Reply:
Worth noting that the mechanics of the hashed wheel timer are exactly the same - I suspect this change will actually hurt performance some (context switching, non-determinism) but I wanted to at least measure it.